### PR TITLE
Code refactor.

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

cc @KochetovNicolai

### CODE REVIEW

File: src/Interpreters/executeQuery.cpp [hunk 1]
```diff
@@ -721,22 +722,6 @@ BlockIO executeQuery(
...
-        bool allow_processors)
-{
-    BlockIO res = executeQuery(query, context, internal, stage, may_have_embedded_data);
-
-    if (!allow_processors && res.pipeline.initialized())
-        res.in = res.getInputStream();
-
-    return res;
-}
-
```
Comment (KochetovNicolai) [inline src/Interpreters/executeQuery.cpp:739]: I think we cannot remove it because of backward compatibility.
Tests should fail

Comment (amosbird) [reply src/Interpreters/executeQuery.cpp:739]: But there is no callsite of this function.

Comment (KochetovNicolai) [reply src/Interpreters/executeQuery.cpp:739]: There is in code used by side projects.

Comment (alexey-milovidov) [reply src/Interpreters/executeQuery.cpp]: It's better to fix side projects than to keep unused garbage in our codebase.

**Review by KochetovNicolai — APPROVED** [review]

**Pull request merged**